python/cudnn/moe_ep/_megamoe.py

```
@@ -0,0 +1,485 @@
+    return _state["modules"]
+
+
+def _ensure_single_rank_dist():
+    """The hybrid layer's comm needs an initialized process group even for
+    ``ep_size=1``; bootstrap a world-1 NCCL group like the megamoe tests."""
+
+    if dist.is_available() and dist.is_initialized():
+        return
+    import tempfile
+
+    store_path = tempfile.mktemp(prefix="cudnn_moe_ep_pg_")
```

Ruff (0.16.0)

[error] Use of insecure and deprecated function (`mktemp`)

(S306)